### ISSUE TITLE
scale factor in inference can be a tensor of dimensions matching random variable, not just a scalar

### ISSUE THREAD

**Issue by dustinvtran** [blei-lab/edward#388]
from discussion with @bayerj at nips. this is useful for applying various masks during the computation of gradients/objectives. the change is also easy i think: move the scale factor into the `tf.reduce_sum`s.
